### ISSUE TITLE
BadRequestError: checks.state argument is missing

### ISSUE THREAD

**Issue by TimothyMerlin** [auth0/express-openid-connect#191]
In production, after users have successfully logged in (this is logged in Auth0), they are sometimes presented with a **Bad Request Error**.

The application logs reads this:
`2021-02-13T17:55:25.757970+00:00 app[web.1]: BadRequestError: checks.state argument is missing
2021-02-13T17:55:25.757980+00:00 app[web.1]:     at /app/node_modules/express-openid-connect/middleware/auth.js:109:31
2021-02-13T17:55:25.757980+00:00 app[web.1]:     at processTicksAndRejections (internal/process/task_queues.js:93:5)`

-  This error never happens on localhost.
- I'm using heroku without a SSL Certificate.
- The browser also shows this issue, which might be related: _Indicate whether to send a cookie in a cross-site request by specifying its SameSite attribute_

### Reproduction

```javascript
const {
    auth,
    requiresAuth
} = require('express-openid-connect');
const {
    response
} = require('express');
require('dotenv').config();

const app = express();
const port = process.env.PORT || 3000
app.listen(port, () =>
    console.log(`Starting server at ${port}`)
);
app.use(express.static('public'));
app.use(express.json({
    limit: '1mb'
}));

//Authentication
const config = {
    authRequired: false,
    auth0Logout: true,
    secret: process.env.SECRET,
    baseURL: process.env.BASE_URL,
    clientID: process.env.CLIENT_ID,
    issuerBaseURL: process.env.ISSUER_BASE_URL
};

// auth router attaches /login, /logout, and /callback routes to the baseURL
app.use(auth(config));

// req.isAuthenticated is provided from the auth router
app.get('/', (req, res) => {
    res.send(req.oidc.isAuthenticated() ? 'Logged in' : 'Logged out');
});

```

### Environment

-   "express-openid-connect": "^2.2.1",

**Comment by adamjmcgrath** [comment]
Hi @TimothyMerlin 

> I'm using heroku without a SSL Certificate.

If you're running your application over `http:` in production and getting intermittent `checks.state argument is missing` errors. then you are almost certainly running into this https://github.com/auth0/express-openid-connect/blob/master/FAQ.md#im-getting-the-warning-using-form_post-for-response_mode-may-cause-issues-for-you-logging-in-over-http-on-localhost 

You can confirm this by performing a login and taking longer than 2 minutes to complete the login on the authorization server.

The obvious solution to this is to run your production server over `https:`

**Comment by TimothyMerlin** [comment]
Thank you very much for your quick reply.

The bad request error that I've described hasn't happened in either http://localhost or https://localhost. 
It only occurs when I push the project to heroku and then try to log in. 

I think I just figured out why that happened thanks to your reply: I've always logged into my app with http://DOMAIN. This resulted in an error in the request due to the problem you explained. The callback then directed me to https://DOMAIN. Logging in then no longer resulted in the error.
